src/bufr/DataObject.h

```diff
@@ -172,6 +225,48 @@ namespace Ingester
             return var;
         };
 
+        /// \brief Makes an new dimension scale using this data objet as the source
+        /// \param name The name of the dimension variable.
+        /// \param dimIdx The idx of the data dimension to use.
+        std::shared_ptr<DimensionDataBase> createDimensionFromData(const std::string& name,
+                                                                   std::size_t dimIdx) const final
+        {
+            auto dimData = std::make_shared<DimensionData<T>>(getDims()[dimIdx]);
+            dimData->dimScale = ioda::NewDimensionScale<T>(name, getDims()[dimIdx]);
+
+            std::copy(data_.begin(),
+                      data_.begin() + dimData->data.size(),
+                      dimData->data.begin());
+
+            // Validate this data object is a valid (has values that repeat for each frame
+            for (size_t idx = 0; idx < data_.size(); idx += dimData->data.size())
+            {
+                if (!std::equal(data_.begin(),
+                                data_.begin() + dimData->data.size(),
+                                data_.begin() + idx,
+                                data_.begin() + idx + dimData->data.size()))
+                {
+                    std::stringstream errStr;
+                    errStr << "Dimension " << name << " has an invalid source field. ";
+                    errStr << "The values don't repeat in each sequence.";
```

Comment:
Suggest not putting apostrophe `don't` in the comment, if someone opens the log in vi/emacs, it might highlight a lot of text thinking it is a string that is unclosed